module/synchronization/core_rapid_test.go

```diff
@@ -77,35 +93,78 @@ func (r *rapidSync) HandleByID(t *rapid.T) {
 	if r.idRequests[b.ID()] == 1 {
 		r.idRequests[b.ID()] = 0
 	}
-	// we eagerly remove height requests
+	// we eagerly remove height requests for blocks we receive
 	r.heightRequests[b.Height] = 0
 }
 
 // Check runs after every action and verifies that all required invariants hold.
 func (r *rapidSync) Check(t *rapid.T) {
-	for k, v := range r.idRequests {
-		if v == 1 {
-			s, ok := r.core.blockIDs[k]
-			require.True(t, ok)
-			assert.True(t, s.WasQueued(), "ID %v was expected to be Queued and is %v", k, s.StatusString())
-		} else if v == 0 {
-			s, ok := r.core.blockIDs[k]
-			// if a block is known with 0 pendings, it's because it was received
-			if ok {
-				assert.True(t, s.WasReceived(), "ID %v was expected to be Received and is %v", k, s.StatusString())
+	// we collect the received blocks as determined above
+	var receivedBlocks []flow.Header
+	// we also collect the pending blocks
+	var activeBlocks []flow.Header
+
+	// we check the validity of our pushdown automaton for ID requests and populate activeBlocks / receivedBlocks
+	for id, count := range r.idRequests {
+		s, foundID := r.core.blockIDs[id]
+
+		block, foundBlock := findHeader(r.store, func(h flow.Header) bool {
+			return h.ID() == id
+		})
+		require.True(t, foundBlock, "incorrect management of idRequests in the tests: all added IDs are supposed to be from the store")
+
+		if count == 1 {
+			require.True(t, foundID, "ID %v is supposed to be known, but isn't", id)
+
+			assert.True(t, s.WasQueued(), "ID %v was expected to be Queued and is %v", id, s.StatusString())
+			assert.False(t, s.WasReceived(), "ID %v was expected to be Queued and is %v", id, s.StatusString())
+			activeBlocks = append(activeBlocks, *block)
+		} else if count == 0 {
+			if foundID {
+				// if a block is known with 0 pendings, it's because it was received
+				assert.True(t, s.WasReceived(), "ID %v was expected to be Received and is %v", id, s.StatusString())
 			}
+			receivedBlocks = append(receivedBlocks, *block)
```

Comment:
should this go inside the if statement?

If a block has 0 pendings and is not known, does this signal an error in the testing logic?